src/theory/strings/theory_strings_type_rules.cpp

```diff
@@ -25,6 +25,20 @@ namespace cvc5::internal {
 namespace theory {
 namespace strings {
 
+bool isMaybeStringLike(const TypeNode& tn)
+{
+  if (tn.isString())
+  {
+    return true;
+  }
+  return tn.isMaybeKind(Kind::SEQUENCE_TYPE);
+}
+
+bool isMaybeInteger(const TypeNode& tn)
```

Comment:
I imagine you have this function also for arithmetic. If possible it would be better to have one copy of it (preferrably in arithmetic and not here), and call it from here.